src/event/handlers/DiscordMessageLinkHandler.ts

```diff
@@ -8,21 +8,23 @@ class DiscordMessageLinkHandler extends EventHandler {
 	}
 
 	async handle(message: Message): Promise<void> {
-		const messageRegex = /https:\/\/(ptb\.)?discord(app)?\.com\/channels\//gm;
-		const linkIndex = message.content.search(messageRegex);
+		const messageRegex = /https:\/\/(?:ptb\.)?discord(?:app)?\.com\/channels\/\d+\/\d+\/\d+/gm;
+		const matches = message.content.matchAll(messageRegex);
 
-		if (linkIndex === -1) return;
+		for (const match of matches) {
+			const index = match.index;
 
-		const wordIndex = message.content.slice(0, linkIndex).split(" ").length - 1;
-		const linkContent = message.content.split(" ")[wordIndex];
+			if (index !== undefined) {
+				let link = match[0];
 
-		if (linkContent.startsWith("<") && linkContent.endsWith(">")) return;
+				if (message.content.charAt(index - 1) !== "<" || message.content.charAt(index + link.length) !== ">") {
+					link = link.replace(/app/, "").replace(/ptb\./, "");
+					const messagePreviewService = MessagePreviewService.getInstance();
```

Comment:
I think it makes sense to declare variables at the top of the block when possible.

Reply:
This point i disagree with. You generally want to declare variables in the lowest scope possible, it makes reasoning about your code a lot simpler.
- It's closer to it's usage so it's easier to understand
- It will leave scope when it's actually meant to rather than lingering around for longer than it should (includes added GC bonus since it can be collected quicker, but in our case that probably doesn't really matter)

Reply:
I personally think it makes the code look messier, but if it has performance benefits I'll allow it.